atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -852,16 +871,30 @@ protected static int getRequestHintToKvStore(int userRequestedSize) {
         return preFilterBatchSize;
     }
 
-    private static Iterator<RowResult<byte[]>> mergeInLocalWritesRows(
+    private Iterator<RowResult<byte[]>> mergeInLocalWritesRows(
             Iterator<RowResult<byte[]>> postFilterIterator,
             Iterator<RowResult<byte[]>> localWritesInRange,
-            boolean isReverse) {
+            boolean isReverse,
+            TableReference tableReference) {
         Ordering<RowResult<byte[]>> ordering = RowResult.getOrderingByRowName();
         Iterator<RowResult<byte[]>> mergeIterators = IteratorUtils.mergeIterators(
                 postFilterIterator, localWritesInRange,
                 isReverse ? ordering.reverse() : ordering,
                 from -> RowResults.merge(from.lhSide, from.rhSide)); // prefer local writes
-        return RowResults.filterDeletedColumnsAndEmptyRows(mergeIterators);
+
+        Iterator<RowResult<byte[]>> purgeDeleted = Iterators.transform(mergeIterators,
+                createFilterColumnValues(tableReference));
+        return Iterators.filter(purgeDeleted, Predicates.not(RowResults.<byte[]>createIsEmptyPredicate()));
+    }
+
+    private Function<RowResult<byte[]>, RowResult<byte[]>> createFilterColumnValues(TableReference tableReference) {
+        return unfilteredRow -> {
+            SortedMap<byte[], byte[]> filteredColumns =
+                    Maps.filterValues(unfilteredRow.getColumns(), Predicates.not(Value.IS_EMPTY));
```

Comment:
out of curiosity, perhaps for later, is it worth unifying `Value.IS_EMPTY` and `Value::isTombstone`? i.e. picking one over the other, feels like the former is pre-java8 style

also, why do we always deal with `byte[]` here, as opposed to some useful safe wrapper like `Value`? This isn't really relevant to this PR

Reply:
1. I'd prefer the latter as long as it isn't used elsewhere outside of our control.
2. Probably because `Value` has a timestamp which a good chunk of the code here isn't directly interested in?

Reply:
1) I had a quick look in PG and it's not.
2) fair enough, I suppose it's the raw byte arrays that make my eyes crawl aha, but it's okay

Reply:
moved everything to use `Value::isTombstone`, and removed `Value.IS_EMPTY`